src/classes/RD2_RecurringDonations_TDTM.cls

```diff
@@ -139,11 +176,71 @@ public class RD2_RecurringDonations_TDTM extends TDTM_Runnable {
     */
     private void buildNewSchedules(List<npe03__Recurring_Donation__c> rds) {
 
-        RD2_ScheduleService scheduleService = new RD2_ScheduleService();
-
         for (npe03__Recurring_Donation__c rd: rds) {
             dmlActions.objectsToInsert.addAll(scheduleService.getNewSchedules(rd));
         }
     }
 
+    /***
+    * @description Handles setting next donation date for updated recurring donations
+    * @param rds Trigger.new records
+    * @param oldRds Trigger.old records
+    */
+    private void setNextDonationDateBeforeUpdate(List<npe03__Recurring_Donation__c> rds, List<npe03__Recurring_Donation__c> oldRds) {
+
+        Map<Id, npe03__Recurring_Donation__c> mapOldRecurringDonationsByIds = new Map<Id, npe03__Recurring_Donation__c>();
+        mapOldRecurringDonationsByIds.putAll(oldRds);
+
+        Map<Id, npe03__Recurring_Donation__c> filteredRecurringDonations = new Map<Id, npe03__Recurring_Donation__c>();
+        for (npe03__Recurring_Donation__c rd : rds) {
+            npe03__Recurring_Donation__c oldRd = mapOldRecurringDonationsByIds.get(rd.Id);
+
+            if (scheduleService.isScheduleUpdateNeeded(rd, oldRd)) {
+                filteredRecurringDonations.put(rd.Id, rd);
+            }
+        }
+
+        for (npe03__Recurring_Donation__c rd : filteredRecurringDonations.values()) {
+            rd.npe03__Next_Payment_Date__c = scheduleService.getNextDonationDate(rd);
+        }
+    }
+
+    /***
+    * @description Handles schedule modification/creation after modification of recurring donation
+    * @param rds Trigger.new records
+    * @param oldRds Trigger.old records
+    */
+    private void synchronizeSchedulesAfterRDUpdate(List<npe03__Recurring_Donation__c> rds, List<npe03__Recurring_Donation__c> oldRds) {
+
+        Map<Id, npe03__Recurring_Donation__c> mapOldRecurringDonationsByIds = new Map<Id, npe03__Recurring_Donation__c>();
+        mapOldRecurringDonationsByIds.putAll(oldRds);
```

Comment:
I recommend moving this block of lines (215 to 225) into a new method getRDsNeedingScheduleUpdate() that returns a list<RD>. That would make this a bit cleaner to read through.